### ISSUE TITLE
Mimic GitHub's editor UI for Backdrop's community forum

### ISSUE THREAD

**Issue by sutibun** [backdrop-ops/backdropcms.org#109]
I began using GitHub this month for the first time (to support Bd) and one thing I've found is how nice the editor UI. (uh... where you type in stuff)

What I like:

- supports Markdown natively
- there's a "Preview" tab allowing you to verify your formatting
- you can easily edit your published comment
- how easily it is to upload images
- use "@" to call attention to someone

I know Bd doesn't currently have a community forum but, when it does, I'd like to request the editor box thingy mimics the one here. It's almost a pleasure to use.

**Comment by quicksketch** [comment]
Hi @sutibun, thanks and all points well made. Backdrop itself (as a product) I think that a rich-text editor is the right move. However, for Backdrop's community (which will mostly consist of developers), I completely agree. On the API website we use BUEditor, which is an assistive HTML toolbar. But we can integrate better with Backdrop now that we support native image handling.

Thanks for making this issue in the right queue as well. :+1:

**Comment by sutibun** [comment]
@quicksketch Yup, I was only referring to the community site. The RTE is the best default choice for the actual CMS that people will use.

Although, I suppose the community forum can provide different editors for users to choose... but maybe something to consider once we get there. I just wanted to get the request in.

I do wonder if GitHub's editor is open source. Maybe we can "hijack" it.

PS: A little weird having the backdropcms.org issues elsewhere and not easily accessible.

**Comment by klonos** [comment]
Doesn't CKeditor already support configurable markdown through plugins or something (so we can create our own or copy the GitHub flavor one)?

https://github.com/hectorguo/CKEditor-Markdown-Plugin
http://ckeditor.com/addon/markdown

![ckeditor-markdown_addon](https://cloud.githubusercontent.com/assets/2423362/10067302/bd1ccd58-62de-11e5-9100-1e04cd310f5f.gif)

**Comment by sutibun** [comment]
I couldn't even install the Bd markdown module
